packages/wrangler/src/__tests__/pages/project-upload.test.ts

```diff
@@ -516,6 +516,67 @@ describe("pages project upload", () => {
 	`);
 	});
 
+	it("should handle a very large number of assets", async () => {
+		const assets = new Set<string>();
+		// Create a large number of asset files to upload
+		for (let i = 0; i < 10_019; i++) {
+			const path = `file-${i}.txt`;
+			const content = `contents of file-${i}.txt`;
+			assets.add(content);
+			writeFileSync(path, content);
+		}
+
+		mockGetUploadTokenRequest(
+			"<<funfetti-auth-jwt>>",
+			"some-account-id",
+			"foo"
+		);
+
+		// Accumulate multiple requests then assert afterwards
+		const uploadedAssets = new Set<string>();
+		msw.use(
+			http.post("*/pages/assets/check-missing", async ({ request }) => {
+				const body = (await request.json()) as { hashes: string[] };
+
+				return HttpResponse.json(
+					{
+						success: true,
+						errors: [],
+						messages: [],
+						result: body.hashes,
+					},
+					{ status: 200 }
+				);
+			}),
+			http.post<never, UploadPayloadFile[]>(
```

Comment:
how many times do we expect this endpoint to be called? should we assert that as well?

Reply:
There are other tests that check how the bucketing works and exactly what requests are made.
This test is purely looking at whether the correct number of assets arrive at the server or not.